src/parsehdr.c

```diff
@@ -298,6 +336,7 @@ cr_package_from_header(Header hdr,
         rpmtdFreeData(filenames);
         rpmtdFreeData(fileflags);
         rpmtdFreeData(filemodes);
+        rpmtdFreeData(filedigests);
     }
 
     rpmtdFree(dirnames);
```

Comment:
I think there is a memory leak of the `filedigests`.
I believe we should also do `rpmtdFree(filedigests);` here.

Reply:
yes, done